### PR TITLE
[Optimizers] Refactored the optimizers in their own directory.

### PR DESCRIPTION
**Pull request by eaplatanios**

### CODE REVIEW

**Comment by eaplatanios** [comment]
How about the current split? I feel "adaptive" is too general in the sense that most optimizers are adaptive in some way, whereas "momentum-based" is a bit more specific to these methods. Also, I used simply "SGD" for the other two, as they are SGD variants. I also feel "gradient descent" is too general as they are all effectively gradient descent methods. Please let me know what you think.

**Comment by rxwei** [comment]
Sounds good!

**Comment by rxwei** [comment]
Maybe rename `MomentumBasedOptimizers.swift` to `MomentumBased.swift`.

**Comment by eaplatanios** [comment]
Done. :)

File: Sources/TensorFlow/Optimizers/SGD.swift [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        if nesterov {
+            model.move(along: momentum * velocity - direction * learningRate)
+        } else {
+            model.move(along: velocity)
+        }
+    }
+}
+
+/// A Riemann manifold stochastic gradient descent (SGD) optimizer.
+public class RiemannSGD<Model: Differentiable>: Optimizer
```
Comment (rxwei) [inline Sources/TensorFlow/Optimizers/SGD.swift]: We can remove `RiemannSGD` at some point since `SGD` is essentially a functionally richer `RiemannSGD` now.
